Change certificate PEM files to PEM format

lucasvdp opened the pull request:
- Removed the quotes and comments out of the PEM files that should not be there.
- Automatically generate hex files that can be included.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nrfconnect/sdk-nrf?pullRequest=13716) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/nrfconnect/sdk-nrf?pullRequest=13716) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nrfconnect/sdk-nrf?pullRequest=13716) it.</sub>

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13716/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13716/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-nrf-iot_lwm2m | X | | |
| test-fw-nrfconnect-nrf-iot_samples | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-13716/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13716
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

jtguggedal commented:
This is a nice improvement! I think you can also move `Amazon root CA 1` from the download sample into the common certs folder. And the common certs folder should probably be documented somewhere, though not sure quite where to put it

lucasvdp commented:
- I moved the certificate from the download sample as well.
- For now I undid the changes to the mqtt sample, with the different overlays this requires a lot more changes. I suggest to limit this PR to only the root certificates.
- If someone has suggestions about what to document where for the common certificate folder let me know.

lemrey commented:
This is fine, but I remember I had discussed moving the certs to a common folder some time ago, and one argument I found against doing that is that it becomes a bit harder to know whether a certificate is still used or not. If you modify a sample to update the certificate it is using, you don't know right away what to do with the old certificate. You can't just remove it, because there might other samples still using it, so you need to look at the other TLS samples too (which now are in two folders, `samples/cellular/` and `samples/net/`).

On the other hand, keeping the certificate next to each sample makes that task easier, and it avoids creating a `common` folder under `samples/` which I don't like. There is a risk that others might want to put things there, which are not certificates. If we do end putting the certificates in one common folder, I argue we it should be called, e.g. `samples/certificates` so that there is no ambiguity as to what can be put there.

What are the disadvantages to using a dedicated folder beside each sample to store the certificates, beyond have more files checked in `git` ?

lucasvdp commented:
I got the request for putting the certificates in a common place, while I was touching them anyway. But don't really care either way myself.
It is nice to keep the sample folders smaller and distinguish between files that might need editing by the user and files that should never be changed.
On the other hand which certificate to use is something they likely need to change and just having it in the sample itself does emphasise that.

The common folder already existed, I just added the certificates folder inside there.

gregersrygg reviewed (approved):
This looks like a nice improvement to me. Regarding where to document it, I think it would be nice to document this in each sample where it is used. It's not necessarily obvious to the user of the sample that the certificate is included using CMakeLists.txt.
This section should also be updated: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/modem/modem_key_mgmt.html#certificates

lemrey commented:
I think we should keep the certificates beside each sample, I do not see any advantages putting them all in one folder.

I could see a point in doing that if we had 20 samples, connecting to the same server with the same certificate.
Then if that server changed its certificate, you'd only need to update the certificate in one place rather than 20.
But right now we have only 4 samples which need a certificate, and they don't even all connect to the same server.

Plus I dislike "common" folders, because they tend to become dumping grounds for things people don't know where to put.

lucasvdp commented:
Thanks,
- I removed the second commit with the certificate move to a common folder.
- Added a few lines of documentation to the samples.
- Changed the modem_key_mgmt.rst document.

jorgenmk reviewed (approved):
LGTM, tested working on https_client and download samples

divipillai commented on samples/cellular/ciphersuites/README.rst:
```diff
@@ -36,6 +36,8 @@ Obtaining a certificate
 
 The sample connects to ``www.example.com``, which requires an X.509 certificate.
 This certificate is provided in the :file:`samples/cellular/https_client/cert` folder.
```
```suggestion
This certificate is provided in the :file:`samples/cellular/ciphersuites/cert` folder.
```

divipillai commented on samples/cellular/ciphersuites/README.rst:
```diff
@@ -36,6 +36,8 @@ Obtaining a certificate
 
 The sample connects to ``www.example.com``, which requires an X.509 certificate.
 This certificate is provided in the :file:`samples/cellular/https_client/cert` folder.
+It is automatically converted to a hex file in :file:`CMakeLists.txt`.
+This hex file is included in :file:`main.c` where it is provisioned to the modem.
```
```suggestion
It is automatically converted to a hex file in :file:`CMakeLists.txt`.
This hex file is included in :file:`main.c`, where it is provisioned to the modem.
```

divipillai commented on samples/net/https_client/README.rst:
```diff
@@ -36,6 +36,8 @@ Obtaining a certificate
 The sample connects to ``www.example.com``, which requires an X.509 certificate.
 This certificate is provided in the :file:`samples/net/https_client/cert` folder.
+It is automatically converted to a hex file in :file:`CMakeLists.txt`.
+This hex file is included in :file:`main.c` where it is provisioned to the modem.
```
```suggestion
This hex file is included in :file:`main.c`, where it is provisioned to the modem.
```
